dora/core/server/worker/src/main/java/alluxio/underfs/BlockMasterSyncHelper.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.worker.block;
```

Comment:
The package is `alluxio.worker.block` but the file is in `src/main/java/alluxio/underfs`. How does this get compiled?

Reply:
So, when doing merge, this file was added into the wrong place?

Reply:
`git cherry-pick` only figures out "oh your target directory has been moved, so i'd better move this file to the 'correct' location", so the target is moved together with its peers and ends up in `underfs`. `git` does not know the package should be renamed as well, hence the stale package name.

This compiles, though, because I think java package name in the code only affects visibility like `protected`? It's not a strong requirement that the package name aligns with the directory name?

Despite that being said, thanks for finding this out! I'll take a quick look at the code and see if it was simply put into an incorrect location (git tried to infer location but made a mistake).

Reply:
> java package name in the code only affects visibility like protected? It's not a strong requirement that

astonishing.